### DISCUSSION TITLE
Bookmarks and Filtering Input Boxes

### DISCUSSION THREAD

**Discussion by hernandito** [coderaiser/cloudcmd#364] [Ideas]
Hi...

I have two ideas that I think would be good improvements.

**BOOKMARKS:**
It would be great if there was a way to bookmark specific folders, and then show you all the bookmarks as buttons at the top of the page. The bookmarked page would load depending on which panel you have active. The goal would be to get to the folder with only one click.

I tried manually adding some html code at the top web page in ` /usr/src/app/html/index.html ` But my edits do not show up. Perhaps it is loading this from someplace else. I even created a new docker repository with my edits. If the bookmark's idea is too complex to implement, to which file would I manually add code to include shortcuts on the main page?

![image](https://user-images.githubusercontent.com/787317/148271258-3b7fbb61-2f35-4968-987f-8f7aa5e93a8a.png)

**FILE FILTERING:**
It would be great to have an input box where once you start typing in there, it would filter the files and folder in the current panel by your search. I have some VERY large directories and it would be great to filter by those. I have seen live filtering boxes that filter contents as you start typing... even if what you are typing is in the middle of the file name. For example If you have a file/folder called "**Red Apple**" and you start typing the word "**apple**" it would highlight these files/folders.

I see that when in a panel, and you type a letter on the keyboard, the list will jump to that letter. This is great...but it does NOT work when I am browsing on my iPad. The iOS keyboard is not active and I cannot type anything. This is why I thought an input box would activate the keyboard.

If doing the filtering idea I describe above is too complex, do you have an idea what I could do to make the keyboard appear and the panel will respond to the letter I type? 

My main purpose for Cloud Commander is to be able to do file operations when I am away from my desktop computer.

Please let me know your thoughts.

Thank you.

**Comment by coderaiser** [comment]
### BOOKMARKS
> It would be great if there was a way to bookmark specific folders, and then show you all the bookmarks as buttons at the top of the page. The bookmarked page would load depending on which panel you have active. The goal would be to get to the folder with only one click.

Here is how `buttons` implement right now:
- [`/client/listeners/index.js`](https://github.com/coderaiser/cloudcmd/blob/v15.9.13/client/listeners/index.js#L111-L144);
- [`/html/index.html`](https://github.com/coderaiser/cloudcmd/blob/v15.9.13/html/index.html#L21-L35);

The best way to achieve this goal would be to use [User Menu](http://cloudcmd.io/#user-menu). You can write any script you want with any modifications of web page.

### FILE FILTERING

- in default mode each letter you right will select file (but only for the first latter), then select next file with name started from the same latter.
- in [**VIM**](http://cloudcmd.io/#vim) mode press slash `/` and then type name of a file you need, then press `Enter`, cursor will highlight the file you need

> I see that when in a panel, and you type a letter on the keyboard, the list will jump to that letter. This is great...but it does NOT work when I am browsing on my iPad. The iOS keyboard is not active and I cannot type anything. This is why I thought an input box would activate the keyboard.

Looks like **VIM** is what you needed.

**Comment by hernandito** [comment]
Hi Guys. Thanks again for the help building a new docker. In my branch, I have made changes to the UI as CSS edits. 

![image](https://user-images.githubusercontent.com/787317/150024984-183c0a21-85a1-4010-8739-6f4d63d8b09e.png)

One other thing you will notice is that I have added buttons to the UI. These buttons are shortcuts to various different folders. My programing skills are not advanced to the point that I can do this as a dynamic user configurable way. I hard coded these in the `index.html` file. @coderaiser I tried following your guide on using the User Menu, but I could not even figure that out. I also like the capability of NOT having to activate the user menu to then select where I want to go. If I had my way, there would be a file that gets stored in the` /root` folder and each user would add the bookmarks they want. 

To take it even further,perhaps there would be a row of buttons at the top of each pane. Clicking the button, would load the folder in that pane only. Something like below photoshopped image.

![image](https://user-images.githubusercontent.com/787317/150026184-e40450a8-06fa-4492-b8b1-17b870bf7d19.png)

As I mentioned, sadly this is way beyond my programming skills. 

Thank you again.

**Reply by coderaiser** [reply]
This is looks very good!
